core/accounting.go

```diff
@@ -6,67 +6,85 @@ import (
 	"time"
 )
 
-// Use transcodeLoopTimeout as TTL for a ManifestID's credit balance
-// We should prevent the use of global state if they are not constants
-
-// Accounts holds credit balances on a per-stream basis
-type Accounts struct {
-	Balances map[ManifestID]balance
-	Mtx      sync.RWMutex
+// Balances holds credit balances on a per-stream basis
+type Balances struct {
+	balances map[ManifestID]*balance
+	mtx      sync.RWMutex
+	ttl      time.Duration
+	quit     chan struct{}
 }
 
 type balance struct {
-	LastUpdate time.Time // Unix time since last update
-	Amount     *big.Rat  // Balance represented as a big.Rat
+	lastUpdate time.Time // Unix time since last update
+	amount     *big.Rat  // Balance represented as a big.Rat
 }
 
-// Credit adds an an amount to the balance for a ManifestID
-func (a *Accounts) Credit(id ManifestID, amount *big.Rat) {
-	a.Mtx.Lock()
-	defer a.Mtx.Unlock()
-	if a.Balances[id].Amount == nil {
-		a.Balances[id] = balance{Amount: big.NewRat(0, 1)}
+// NewBalances returns a pointer to a Balance type with an initialized map
```

Comment:
I think its clear from the function signature that we're returning a `*Balances` type so maybe we can just say something like: "NewBalances creates a Balances instance with the given ttl`

Reply:
Adopted your comment in https://github.com/livepeer/go-livepeer/pull/956/commits/789195926a3d4628b3bc3e6acec20f4b78035e68